### DISCUSSION TITLE
Root wildcards and nesting

### DISCUSSION THREAD

**Discussion by ghost** [tokio-rs/axum#1561] [Q&A] [answered]
Hello. How to serve statics files as the root url, and the API from /api prefix ? In this form, it gives an error in runtime about overlapping routes:

```
    Router::new()
        .nest("/api", api)
        .route("/", get(get_index))
        .route("/*file", get(get_static))
```

**Comment by davidpdrsn** [comment, marked as answer]
You need to use `Router::fallback` for serving the static files. See https://github.com/tokio-rs/axum/blob/main/examples/static-file-server/src/main.rs#L80-L89

**Reply by ghost** [reply]
Thank you!
